rust-runtime/aws-smithy-types/src/timeout/api.rs

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::tristate::TriState;
+use std::time::Duration;
+
+/// API timeouts used by Smithy `Client`s
+#[non_exhaustive]
+#[derive(Clone, PartialEq, Default, Debug)]
+pub struct Api {
+    /// A limit on the amount of time it takes for the first byte to be sent over an established,
+    /// open connection and when the last byte is received from the service for a single attempt.
+    /// If you want to set a timeout for an entire request including retry attempts,
+    /// use [`Api::call_attempt`] instead.
+    call: TriState<Duration>,
+    /// A limit on the amount of time it takes for request to complete. A single request may be
+    /// comprised of several attempts depending on an app's [`RetryConfig`](crate::retry::RetryConfig). If you want
+    /// to control timeouts for a single attempt, use [`Api::call`].
+    call_attempt: TriState<Duration>,
```

Comment:
Seems like these doc comments should go on the setters in the impl since they won't show up in the docs on these private fields.